src/storage/data_table.cpp

```diff
@@ -1334,9 +1335,7 @@ bool DataTable::ScanCreateIndex(CreateIndexScanState &state, DataChunk &result,
 void DataTable::AddIndex(unique_ptr<Index> index, const vector<unique_ptr<Expression>> &expressions) {
```

Comment:
Not sure if I understand correctly, but is this `AddIndex` method still used? If not, can it be reworked to only add a completed index, or be removed?

From what you told me, I thought all of this logic is now handled by `PhysicalCreateIndex`

Reply:
Ah yes I completely missed that. My code currently only runs for the `CREATE INDEX idx ON table USING art(id)` statement. 

We still use the `AddIndex` for creating tables with constraints directly (`TableCatalogEntry` constructor), e.g. `CREATE TABLE test (id INTEGER PRIMARY KEY)`. Maybe we can have a chat on how to change that to also use the pipeline?

Reply:
There is no data in a `CREATE TABLE ` statement yet, so we can just add the index here without looping through the data. Removing the `DataTable::AddIndex` should just work.